proxy/http/HttpTransact.cc

```diff
@@ -3641,12 +3641,12 @@ HttpTransact::handle_response_from_parent(State *s)
       return;
     }
 
-    if (s->current.attempts.get() < s->txn_conf->parent_connect_attempts) {
```

Comment:
isn't this an off by one error then? if `s->txn_conf->parent_connect_attempts` is 3 so we can try 3 times and we are on the 3rd attempt but since we don't increment yet, `s->current.attempts.get() = 2` so that conditional would fail so we don't try the 3rd time?

Reply:
Good point. I think the confusion arises from the naming. `s->current.attempts.get()` is actually the retry count, so when `s->current.attempts.get() = 2`, it has retried for 2 times already(3 attempts in total).

The naming confusion is addressed in another PR: https://github.com/apache/trafficserver/pull/9655 .